src/components/ICA/__tests__/ICA.spec.js

```diff
@@ -3,76 +3,82 @@
  * @copyright IBM Security 2019
  */
 
-import { shallow } from 'enzyme';
 import React from 'react';
+import { render } from '@testing-library/react';
 
 import { ICA } from '../../..';
 import { Locales } from '../ICA';
 
-import props from '../_mocks_';
-
-const { label, total, value } = props;
-
-const defaultProps = { label, value };
-
 describe('ICA', () => {
-  let ica;
-  const largeValue = value * total;
-
-  beforeEach(() => {
-    ica = shallow(<ICA {...defaultProps} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(<ICA label="test ICA" total={10} value={5} />, {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    });
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('ICA');
   });
 
-  it('renders', () => {
-    expect(ica).toMatchSnapshot();
+  it('should render en dash when `value` is `null`', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={null} />);
+    expect(queryByText('–')).toBeVisible();
   });
 
-  it('renders the null value', () => {
-    ica.setProps({ value: null });
-
-    expect(ica).toMatchSnapshot();
+  it('should render a large value', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={1000000} />);
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('renders a large value', () => {
-    ica.setProps({ value: largeValue });
-
-    expect(ica).toMatchSnapshot();
+  it('should render a subset of values when `value` and `total` are provided', () => {
+    const { getByText } = render(<ICA label="test ICA" value={5} total={10} />);
+    expect(getByText('5').nextSibling).toBe(getByText(/10/i));
```

Comment:
I had a quick question on some of the following unit tests - is there an alternative way to query / verify these without relying heavily on the internal DOM structure (`nextSibling`, `parentNode.children`, etc.)? I worry that if this structure was to change but the logic remains the same, these tests would no longer pass.

Reply:
🤔 I understand your concern, but wanted to point out two things --

1. In this way, it's not so different from the snapshot tests, where if DOM structured changed, they had to be revisited + regenerated (and those tests sufficed for a time 🤷‍♀ )

2. Moreso to your point though -- I think the ICA is different in this respect. We are dealing with a component that displays fractions or percentages. So for example, if the next sibling of the numerator is NOT the denominator, we have a problem -- and that problem could mean that the component is unreadable for screen readers.

Reply:
Sorry, I'm also trying to reinforce my understanding of React Testing Library here at the same time!

That makes sense to me - thanks for clarifying!

Reply:
It's probably overkill though. 🤔  I have an update to remove a lot of this.